bql/planner/data_access.go

```diff
@@ -409,7 +409,12 @@ func simpleFetch(ctx context.Context, gs []storage.Graph, cls *semantic.GraphCla
 // table. The semantic graph clause is also passed to be able to identify what
 // bindings to set.
 func addTriples(ts <-chan *triple.Triple, cls *semantic.GraphClause, tbl *table.Table) error {
+	var lastErr error
 	for t := range ts {
+		if lastErr != nil {
+			// Drain the channel to avoid leaking goroutines.
+			continue
+		}
```

Comment:
maybe do that change where we move this to after the loop and use a brak to get there, instead of continue.

Reply:
It appeared here again just because this branch `time-binding-behavior` is rebased over the not yet merged branch `fix-type-object` of a previous PR (that is passing Go readability now). But the refactor we did using break and moving the draing logic to a moment after the loop is also here too (see commit "Refactor the function "addTriples" to make it clearer" above).
These commits of the previous PR appear again here because of the way GitHub manages the PRs without looking at the previous ones or letting us identify other PRs that must be merged before the current one. It has been annoying me a little too, but I do not know another way at the moment to work around it.